### PR TITLE
fixed and extended email/phone validation

### PR DESCRIPTION
**Pull request by wpietri**

### CODE REVIEW

File: src/generateFormFields.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ export interface FormFields {
 }
 
 export function generateFormFields(step: Step): FormFields {
+	const phoneRegExp = /[+]?[0-9][-.0-9 ]{8,}[0-9]/
```
Comment (imf) [inline src/generateFormFields.tsx:18]: We should probably either:

* Use the NANP regex: ^(?:(?:\+?1\s*(?:[.-]\s*)?)?(?:\(\s*([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9])\s*\)|([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9]))\s*(?:[.-]\s*)?)?([2-9]1[02-9]|[2-9][02-9]1|[2-9][02-9]{2})\s*(?:[.-]\s*)?([0-9]{4})(?:\s*(?:#|x\.?|ext\.?|extension)\s*(\d+))?$
* Ignore all non-digit characters and make sure there are at 10 or 11 digits (and if 11, that the first one is 1.) 
* Allow ( ) around the area code.
* Something else... But the NANP regex seems like a good choice!

And we may want to strip all non-digit characters from what we either persist or transmit to SF.

Comment (wpietri) [reply src/generateFormFields.tsx:18]: I looked at more complex regexs, but intentionally chose something simpler. My thinking is that a large blob of regex like that with no tests is effectively write-only code. I have no idea if it's really better, because it's too complex to understand by reading it. I also had concerns that there might be edge cases where Americans needing medical care might need callbacks at non-American numbers. And lastly, a person intent on entering a fake phone number can just as well enter somebody else's number, so I'm not sure more complex validation solves a real problem. But if you want, I'm glad to paste that in and cross my fingers.

I think just making sure there at least 10 digits would be good, but I don't think that can be expressed as a regex, and since the existing code is using a complex validation framework that I don't know how to write tests for, I'm reluctant to try to add that.

I agree allowing parens is valuable and will add that depending on the path you prefer.

Changing the data before it gets to Salesforce could also be a good thing, but I'd want to check with the SF devs. If there's anything I'd expect a CRM to be good at, it's phone numbers, so it may not be necessary. And I suspect there are cases where stripping formatting makes things worse, so I'd want to discuss possible edge cases too.

Comment (imf) [reply src/generateFormFields.tsx:18]: I agree it's write-only but some quick manual testing should at least validate that it works with a few use cases. Agree with the sentiment though.

And the big regex is something we could put in now, that would allow parentheses.

Comment (imf) [reply src/generateFormFields.tsx:18]: Though to your point, it's basically almost impossible to reason about.

We could strip the result of non-digits and check for length, in the validation check.

In any case, I think the simple one you have is too aggressive, and this opaque one seems to be less so.

Comment (imf) [reply src/generateFormFields.tsx:18]: (I'd feel different if the simple one didn't present obvious false negatives. And for email especially, I want no part of trying to apply complex rules to what ISN'T an email address.)

Comment (wpietri) [reply src/generateFormFields.tsx]: Ok. Going with the one from SO. Which, for whoever comes back to this later, allows some weirdness (eg, 7-digit numbers) and will accept Canadian numbers, but will not allow anybody outside US/CA, American included, to use the form.
